### PR TITLE
DOCSUP-6074: Documented DETACH TABLE PERMANENTLY

### PR DESCRIPTION
**Pull request by gyuton**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #17642

### CODE REVIEW

File: docs/en/sql-reference/statements/detach.md [hunk 1]
```diff
@@ -5,12 +5,30 @@ toc_title: DETACH
 
 # DETACH Statement {#detach}
 
-Deletes information about the ‘name’ table from the server. The server stops knowing about the table’s existence.
+Deletes information about the `name` table from the server. The server stops knowing about the table’s existence.
+
+Syntax:
 
 ``` sql
 DETACH TABLE [IF EXISTS] [db.]name [ON CLUSTER cluster]
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:13]:
```suggestion
DETACH TABLE|VIEW [IF EXISTS] [db.]name [PERMANENTLY] [ON CLUSTER cluster]
```

File: docs/en/sql-reference/statements/detach.md [hunk 2]
```diff
@@ -5,12 +5,30 @@ toc_title: DETACH
 
 # DETACH Statement {#detach}
 
-Deletes information about the ‘name’ table from the server. The server stops knowing about the table’s existence.
+Deletes information about the `name` table from the server. The server stops knowing about the table’s existence.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:8]:
```suggestion
Deletes information about the table or view from the server. The server stops knowing about their existence.
```

File: docs/en/sql-reference/statements/detach.md [hunk 3]
```diff
@@ -5,12 +5,30 @@ toc_title: DETACH
...
+
+Syntax:
 
 ``` sql
 DETACH TABLE [IF EXISTS] [db.]name [ON CLUSTER cluster]
 ```
 
 This does not delete the table’s data or metadata. On the next server launch, the server will read the metadata and find out about the table again.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:16]:
```suggestion
Detaching does not delete the table’s data or metadata. If the table was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table again. If the table was detached `PERMANENTLY`, there will be no automatic recall. 

Whether the table was detached permanently or not, in both cases you can reattach it using the [ATTACH](../../sql-reference/statements/attach.md) query (with the exception of system tables, which do not have metadata stored for them).
```

File: docs/en/sql-reference/statements/detach.md [hunk 4]
```diff
@@ -5,12 +5,30 @@ toc_title: DETACH
...
+
+Syntax:
+
+``` sql
+DETACH TABLE/VIEW [IF EXISTS] [db.]name PERMAMENTLY [ON CLUSTER cluster]
+```
+
+This statement does not delete the table’s data or metadata.
+
+Permanently detached table or view can be reattached with [ATTACH](../../sql-reference/statements/attach.md) query and can be shown with [SHOW CREATE TABLE](../../sql-reference/statements/show.md#show-create-table) query.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:32]:
```suggestion

```

File: docs/en/sql-reference/statements/detach.md [hunk 5]
```diff
@@ -5,12 +5,45 @@ toc_title: DETACH
 
 # DETACH Statement {#detach}
 
-Deletes information about the ‘name’ table from the server. The server stops knowing about the table’s existence.
+Deletes information about the table or view from the server. The server stops knowing about their existence.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:8]:
```suggestion
Deletes information about the table or materialized view from the server. The server stops knowing about their existence.
```

File: docs/en/sql-reference/statements/detach.md [hunk 6]
```diff
@@ -5,12 +5,45 @@ toc_title: DETACH
...
+
+Syntax:
 
 ``` sql
-DETACH TABLE [IF EXISTS] [db.]name [ON CLUSTER cluster]
+DETACH TABLE|VIEW [IF EXISTS] [db.]name [PERMANENTLY] [ON CLUSTER cluster]
 ```
 
-This does not delete the table’s data or metadata. On the next server launch, the server will read the metadata and find out about the table again.
+Detaching does not delete the data or metadata for the table or view. If the table or view was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table/view again. If the table or view was detached `PERMANENTLY`, there will be no automatic recall.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:16]:
```suggestion
Detaching does not delete the data or metadata for the table or materialized view. If the table or view was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table/view again. If the table or view was detached `PERMANENTLY`, there will be no automatic recall. 
```

File: docs/en/sql-reference/statements/detach.md [hunk 7]
```diff
@@ -5,12 +5,45 @@ toc_title: DETACH
...
+**Example**
+
+Query:
+
+``` sql
+CREATE TABLE test ENGINE = Log AS SELECT * FROM numbers(10);
+
+DETACH TABLE test;
+
+SELECT * FROM TEST;
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:35]:
```suggestion
SELECT * FROM test;
```

File: docs/en/sql-reference/statements/detach.md [hunk 8]
```diff
@@ -5,12 +5,45 @@ toc_title: DETACH
...
+Result:
+
+``` text
+Ok.
+
+Ok.
+
+Received exception from server (version 21.3.1):
+Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: Table default.TEST doesn't exist.
+```
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:48]: Then attach the table again:

```sql
ATTACH TABLE test;
SELECT * FROM test;
```

Result:
```text
┌─number─┐
│      0 │
│      1 │
│      2 │
│      3 │
│      4 │
│      5 │
│      6 │
│      7 │
│      8 │
│      9 │
└────────┘
```

File: docs/en/sql-reference/statements/detach.md [hunk 9]
```diff
@@ -5,12 +5,45 @@ toc_title: DETACH
...
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:47]:
```suggestion
Creating a table:

``` sql
CREATE TABLE test ENGINE = Log AS SELECT * FROM numbers(10);
SELECT * FROM test;
```
Result:
```text
┌─number─┐
│               0 │
│               1 │
│               2 │
│               3 │
│               4 │
│               5 │
│               6 │
│               7 │
│               8 │
│               9 │
└───────┘
```
Detaching the table:

```sql
DETACH TABLE test;
SELECT * FROM test;
```
Result:

``` text
Received exception from server (version 21.3.1):
Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: Table default.test doesn't exist.
```

File: docs/en/sql-reference/statements/detach.md [hunk 10]
```diff
@@ -5,12 +5,66 @@ toc_title: DETACH
 
 # DETACH Statement {#detach}
 
-Deletes information about the ‘name’ table from the server. The server stops knowing about the table’s existence.
+Deletes information about the table or materialized view from the server. The server stops knowing about their existence.
```
Comment (filimonov) [inline docs/en/sql-reference/statements/detach.md:8]: It doesn't delete anything. May be smth like "Unregister the table / Makes the server forget about the table"

File: docs/ru/sql-reference/statements/attach.md [hunk 11]
```diff
@@ -10,14 +10,15 @@ toc_title: ATTACH
 -   вместо слова `CREATE` используется слово `ATTACH`;
 -   запрос не создаёт данные на диске, а предполагает, что данные уже лежат в соответствующих местах, и всего лишь добавляет информацию о таблице на сервер. После выполнения запроса `ATTACH` сервер будет знать о существовании таблицы.
 
-Если таблица перед этим была отсоединена (`DETACH`), т.е. её структура известна, можно использовать сокращенную форму записи без определения структуры.
+Если таблица перед этим была откреплена ([DETACH](../../sql-reference/statements/detach.md)), т.е. её структура известна, можно использовать сокращенную форму записи без определения структуры.
```
Comment (filimonov) [inline docs/ru/sql-reference/statements/attach.md:13]: Может быть отключена/подключена?

**Review by tavplubix — APPROVED** [review]

**Pull request merged**